File: dotnet/SeedAudioAISiteKit.cs
namespace SeedAudioAISiteKit;

public sealed record SiteMetadata(
    string Name,
    string Homepage,
    string Description,
    IReadOnlyDictionary<string, string> CanonicalPages,
    IReadOnlyList<string> Tags
);

public static class SeedAudioAI
{
    public const string Homepage = "https://seedaud.io/";

    public static SiteMetadata Site() => new(
        "Seed Audio AI",
        Homepage,
        "Browser-based text-to-speech and AI voice generation workflow.",
        new Dictionary<string, string>
        {
            ["home"] = Homepage,
            ["textToSpeech"] = "https://seedaud.io/text-to-speech/",
            ["pricing"] = "https://seedaud.io/pricing/",
            ["safety"] = "https://seedaud.io/safety/",
            ["terms"] = "https://seedaud.io/terms/"
        },
        new[] { "text-to-speech", "ai-voice", "voice-generator", "site-kit" }
    );

    public static string Url(string? path = null)
    {
        var clean = (path ?? string.Empty).Trim('/');
        return clean.Length == 0 ? Homepage : $"{Homepage}{clean}/";
    }
}

File: ruby/lib/seed_audio_ai_site_kit.rb
module SeedAudioAISiteKit
  SITE = {
    name: "Seed Audio AI",
    homepage: "https://seedaud.io/",
    description: "Browser-based text-to-speech and AI voice generation workflow.",
    canonical_pages: {
      home: "https://seedaud.io/",
      text_to_speech: "https://seedaud.io/text-to-speech/",
      pricing: "https://seedaud.io/pricing/",
      safety: "https://seedaud.io/safety/",
      terms: "https://seedaud.io/terms/"
    },
    tags: ["text-to-speech", "ai-voice", "voice-generator", "site-kit"]
  }.freeze

  def self.seed_audio_url(path = "")
    clean = path.to_s.gsub(%r{\A/+|/+\z}, "")
    clean.empty? ? SITE[:homepage] : "#{SITE[:homepage]}#{clean}/"
  end
end
